### PR TITLE
container: Add redis + DTO image for DSA.

### PR DESCRIPTION
**Pull request by pingzhaozz**
This PR is to add the container for dsa enabled redis.

### CODE REVIEW

File: cnap/redis/idxd-reset.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+From 07e1137a71021dc02df98b942a260b70ff852ca5 Mon Sep 17 00:00:00 2001
```
Comment (leyao-daily) [inline cnap/redis/idxd-reset.patch]: All these are for redis images build, so maybe we cloud move all of them to ,container/cnap-redis/patches or something. Update the docker image then.

File: helm/redis/templates/deployment.yaml [hunk 2]
```diff
@@ -22,6 +22,9 @@ spec:
         - name: config-volume
           configMap:
             name: {{ include "redis.fullname" . }}
+        - name: devfs
```
Comment (leyao-daily) [inline helm/redis/templates/deployment.yaml]: Only configure this when dsa enabled.

File: cnap/redis/idxd-reset.patch (hunk 1 above)
Comment (pingzhaozz) [reply cnap/redis/idxd-reset.patch]: ok.

File: helm/redis/templates/deployment.yaml (hunk 2 above)
Comment (pingzhaozz) [reply helm/redis/templates/deployment.yaml]: I'll remove these, and move resources to deployment.yaml

**Comment by leyao-daily** [comment]
Thanks for your changes, and 
1. For `redis.conf`, I checked and found just a few changes have been made. Is these changes necessary for redis dsa enabled?
2. Why remove resource and I don't find anything changed in deployment.

**Comment by pingzhaozz** [comment]
> Thanks for your changes, and
> 
> 1. For `redis.conf`, I checked and found just a few changes have been made. Is these changes necessary for redis dsa enabled?
> 2. Why remove resource and I don't find anything changed in deployment.

redis.conf enable the multi io threads for better dsa offloading.
I move the resource field in value.yaml to deployment.yaml because dsa has following resources:
```
        resources:
          limits:
            dsa.intel.com/wq-user-shared: 2
```
To avoid duplicate & override, better to keep 1 location

**Review by leyao-daily — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**